### PR TITLE
Feat/1354  export note to pdf

### PR DESCRIPTION
**Pull request by inkhey**
related to #1354 

## Checkpoints

**For developers**

- [x] If relevant, manual tests have been done to ensure the stability of the whole application and that the involved feature works
- [x] The original issue is up to date w.r.t the latest discussions and contains a short summary of the implemented solution
- [x] Automated tests covering the feature or the fix, have been written, deemed irrelevant (give the reason), or an issue has been created to implement the test (give the link)
- [x] Make sure that:
  - if there are modifications in the Tracim configuration files (eg. `development.ini`), they are documented in `backend/doc/setting.md`
  - any migration process required for existing instances is documented
  - relevant people for these changes are notified
- [x] Original authors of the features included in a multi-feature branch (maintenance fixes -> develop, security fixes -> develop, …) should be part of the reviewers, especially if you encountered merge conflicts.

**For code reviewers**

- [x] The code is clear enough
- [x] If there are FIXMEs in the code, related issues are mentioned in the FIXME
- [x] If there are TODOs, NOTEs or HACKs in code, the date and the developer initials are present

**For testers**

- [x] Manual, quality tests have been done

### CODE REVIEW

File: backend/setup.py [hunk 1]
```diff
@@ -81,6 +81,9 @@
     "webpreview @ git+https://github.com/algoo/webpreview@v1.6.0+algoo",
     # importlib
     "importlib_metadata==4.6.0",
+    # note pdf preview
+    "pypandoc",
+    "weasyprint<53",
```
Comment (inkhey) [inline backend/setup.py:86]: weasyprint 53 and later doesn't work on ubuntu 18.04 and debian buster because it require Pango >= 1.44.0.

Comment (inkhey) [reply backend/setup.py:86]: for information, the erreur obtained:
```
AttributeError: function/symbol 'pango_context_set_round_glyph_positions' not found in library 'libpango-1.0.so.0': /usr/lib/x86_64-linux-gnu/libpango-1.0.so.0: undefined symbol: pango_context_set_round_glyph_positions
Error producing PDF
```

File: backend/tracim_backend/config.py [hunk 2]
```diff
@@ -1024,6 +1037,17 @@ def _check_global_config_validity(self) -> None:
...
+            "RICH_TEXT_PREVIEW__TEMPLATE_PATH", self.RICH_TEXT_PREVIEW__TEMPLATE_PATH
+        )
+        self.check_file_path_param(
+            "RICH_TEXT_PREVIEW__TEMPLATE_PATH", self.RICH_TEXT_PREVIEW__TEMPLATE_PATH, readable=True
+        )
+        self.check_mandatory_param("RICH_TEXT_PREVIEW__CSS_PATH", self.RICH_TEXT_PREVIEW__CSS_PATH)
+        self.check_file_path_param(
+            "RICH_TEXT_PREVIEW__CSS_PATH", self.RICH_TEXT_PREVIEW__CSS_PATH, readable=True
+        )
+
```
Comment (inkhey) [inline backend/tracim_backend/config.py:1050]: This make 2 new files from branding mandatory for all installations: https://github.com/tracim/tracim/issues/5711 I don't really like the fact we have such breaking change forcing sys-administrator to do manual fix but i don't see better solution for now, this feature doesn't make much sense to be optional and having tracim in a partially working state is not a good solution.

@PhilippeAccorsi What do you think about this kind of breaking change ?

File: backend/tracim_backend/lib/rich_text_preview/html_preview.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        content: str,
+        filename: str,
+        default_filename: str,
+        force_download: bool,
+        last_modified,
+        metadata: typing.Dict[str, typing.Any],
+    ):
+
+        with tempfile.NamedTemporaryFile(
+            "w+b", prefix="preview-generator-", suffix=".pdf", delete=False
```
Comment (grignards) [inline backend/tracim_backend/lib/rich_text_preview/html_preview.py:25]:
```suggestion
            "w+b", prefix="tracim-notes-preview-", suffix=".pdf", delete=False
```

File: backend/tracim_backend/applications/content_html_document/controller.py [hunk 4]
```diff
@@ -285,6 +285,70 @@ def set_html_document_status(self, context, request: TracimRequest, hapic_data=N
...
+            config=app_config,
+        )
+        content = api.get_one(hapic_data.path.content_id, content_type=content_type_list.Any_SLUG)
+        default_filename = "{label}.pdf".format(label=content.label)
+        result = api.get_full_pdf_preview_from_html_raw_content(
+            revision=content.revision,
+            filename=hapic_data.path.filename,
+            default_filename=default_filename,
+            force_download=hapic_data.query.force_download,
+            additional_metadata={},
```
Comment (grignards) [inline backend/tracim_backend/applications/content_html_document/controller.py:315]: As `additional_metadata` seems unused I would rather remove it.

File: backend/tracim_backend/lib/utils/utils.py [hunk 5]
```diff
@@ -541,3 +543,18 @@ def validate_data(self, params: Dict[str, Any], json_schema: Dict[str, Any]):
                     "> ".join([str(item) for item in exc.absolute_path]), exc.message
                 )
             ) from exc
+
+
+def date_as_lang(datetime_obj: datetime, locale: str = None, default_locale: str = None):
```
Comment (grignards) [inline backend/tracim_backend/lib/utils/utils.py:548]: Missing type annotation for returned value.

Comment (grignards) [reply backend/tracim_backend/lib/utils/utils.py:548]: And `Optional` for the strings.

**Review by grignards — CHANGES_REQUESTED** [review]

**Review by Haskaris — APPROVED** [review]

File: backend/tracim_backend/lib/core/content.py [hunk 6]
```diff
@@ -713,6 +718,71 @@ def get_one_page_pdf_preview(
...
+        self,
+        revision: ContentRevisionRO,
+        filename: str,
+        default_filename: str,
+        additional_metadata: typing.Dict[str, typing.Any],
+        force_download: bool = None,
+    ):
+        content = revision.node
+        revision_in_context = self.get_revision_in_context(revision)
+        space_name = _("espace {workspace_name}").format(workspace_name=content.workspace.label)
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/tracim_backend/lib/core/content.py]: right

File: backend/doc/branding.md [hunk 7]
```diff
@@ -120,3 +120,17 @@ supported by browsers. These files it will be accessible through links.
...
+
+
+## Customize Note as pdf convertion rendering
+
+Tracim convert Note as PDF using both pandoc and weasyprint tool.
+This process is customizable through 2 files in branding:
+- The css style : `rich_text_preview.css`.
+- The html template : `rich_text_preview.template`.
+
+The first one(`rich_text_preview.css`) is a simple css file. You can use advanced feature for pagination file convertion
```
[the remark this answers is not in the document]
Comment (inkhey) [reply backend/doc/branding.md]: done.

File: backend/tracim_backend/applications/content_html_document/controller.py (hunk 4 above)
Comment (inkhey) [reply backend/tracim_backend/applications/content_html_document/controller.py]: done.

File: backend/tracim_backend/lib/utils/utils.py (hunk 5 above)
Comment (inkhey) [reply backend/tracim_backend/lib/utils/utils.py]: done.

**Pull request merged**